### PR TITLE
note page

### PR DESCRIPTION
**Pull request by kvoncode**
**WARNING** The tree uses **react-router** which needs to be configured in production as described in https://ui.dev/react-router-cannot-get-url-refresh/

Currently in dev environment webpack is configured to redirect all requests to index.html in order to load react-router

Please review and merge

### CODE REVIEW

**Comment by DarthVanger** [comment]
Why is the PR named "note page" if it's not adding a note page? It's only changing the start script to use webpack dev server instead of `server.js`, changes webpack code style, adds a few new libs to `package.json`.

**Comment by kvoncode** [comment]
> Why is the PR named "note page" if it's not adding a note page? It's only changing the start script to use webpack dev server instead of `server.js`, changes webpack code style, adds a few new libs to `package.json`.
> 
> Please rename the PR and I'll merge it.
> I find it weird to add new libs without using them though...

Sorry, my bad I should have provided description that it's not finished yet. I was checking PR correctness. I will request review when PR is ready

**Comment by kvoncode** [comment]
Finally app looks like this
Screenshot 2020-10-23 17:30:31

Please review and merge

File: src/components/LoginPage/LoginPageComponent.js [hunk 1]
```diff
@@ -89,7 +89,9 @@ const Button = css`
 
 const GoogleLoginButton = styled.button`
   ${Button}
-  float: right;
+  position: absolute;
+  right: 2rem;
+  top: 1rem;
```
Comment (DarthVanger) [inline src/components/LoginPage/LoginPageComponent.js:94]: Did we have any problems with styles on login page?
@TomaTomoe pls review

File: src/components/LoginPage/LoginPageComponent.js [hunk 2]
```diff
@@ -115,9 +117,7 @@ const Footer = styled.footer`
 
 export class LoginPageComponent extends Component {
   render() {
-    const authHandler = this.props.isGoogleApiInitialized
-      ? this.props.requestAuthorization
-      : null;
+    const authHandler = this.props.requestAuthorization;
```
Comment (DarthVanger) [inline src/components/LoginPage/LoginPageComponent.js:120]: Why this change?

File: src/components/LoginPage/LoginPageSagas.js [hunk 3]
```diff
@@ -34,6 +33,6 @@ export function* handleAuth() {
 
 export function* loginInit() {
   yield googleApiLoader.load();
-  yield put(googleApiInitializedAction());
```
Comment (DarthVanger) [inline src/components/LoginPage/LoginPageSagas.js:37]: Why this change?

File: src/components/LogoutButton/LogoutButtonComponent.js [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
-import { StyledLogoutButton } from 'components/LogoutButton/LogoutButtonStyles';
+// import { StyledLogoutButton } from 'components/LogoutButton/LogoutButtonStyles';
+import Button from '@material-ui/core/Button';
 
 export class LogoutButtonComponent extends Component {
   render() {
     return (
-      <StyledLogoutButton onClick={this.props.logout}>
+      <Button
+        style={{ color: '#E4E3E4', margin: '1rem' }}
```
Comment (DarthVanger) [inline src/components/LogoutButton/LogoutButtonComponent.js:9]: Is it a styling to implement the new app design or just your custom design to make it prettier?
@TomaTomoe pls review

File: src/components/NoteDetailsButton/NoteDetailsButtonComponent.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import IconButton from '@material-ui/core/IconButton';
+import EditIcon from '@material-ui/icons/Edit';
+
+import { Link } from 'react-router-dom';
+
+const FlexContainer = styled.div`
+  position: absolute;
+  width: 100vw;
+  z-index: 100;
```
Comment (DarthVanger) [inline src/components/NoteDetailsButton/NoteDetailsButtonComponent.js:12]: Why 100?

File: src/components/App/AppComponent.js [hunk 6]
```diff
@@ -1,48 +1,33 @@
...
+        <Route exact path="/">
+          <LoginPageContainer />
+        </Route>
+        <Route exact path="/notes">
+          <NotesPageContainer />
+        </Route>
+        <Route exact path="/note">
+          <NoteDetailsPage />
+        </Route>
+        <Route path="/upload">
```
Comment (DarthVanger) [inline src/components/App/AppComponent.js:24]: I think the url should be uploadS

File: src/components/LoginPage/LoginPageComponent.js (hunk 1 above)
Comment (kvoncode) [reply src/components/LoginPage/LoginPageComponent.js:94]: It adds a little space on top. Generally using I would not recommend using float, instead you can use flex container as complete replacement, or other solutions. In our case we need position button on the right upper corner and it will look nicer with a little margin (space) between edge of the page and button itself.

File: src/components/LoginPage/LoginPageComponent.js (hunk 2 above)
Comment (kvoncode) [reply src/components/LoginPage/LoginPageComponent.js:120]: As you can see in the code **authHandler** is on click handler. Generally you never assign handle function conditionally or at least I can't came up with use case where you will need to do so.

Also **isGoogleApiInitialized** is true when corresponding action is triggered, but the action itself changes no state. Basically it's more complicated way to **console.log("GAPI Initialized"), but the main reason is the reason above, that it serves no function except being a condition where it need not to be.

You can think that we need conditional assignment because we don't want button to use uninitialized code, but the **isGoogleApiInitialized** itself not connected to any code that should be initialized before using, it just exists independently

**requestAuthorization** function itself just makes the same, emitting another action, so it's safe to assign handler to button

File: src/components/LoginPage/LoginPageSagas.js (hunk 3 above)
Comment (kvoncode) [reply src/components/LoginPage/LoginPageSagas.js:37]: It's basically **console.log**, but made with redux and saga

File: src/components/LogoutButton/LogoutButtonComponent.js (hunk 4 above)
Comment (kvoncode) [reply src/components/LogoutButton/LogoutButtonComponent.js:9]: Just some fast styles to see how it can look. Anybody are welcome to fix it or change styles completely

File: src/components/NoteDetailsButton/NoteDetailsButtonComponent.js (hunk 5 above)
Comment (kvoncode) [reply src/components/NoteDetailsButton/NoteDetailsButtonComponent.js:12]: It's fast styles and unfinished styling, I was told not to concentrate too much on styles, so I decided to leave it for others to style. I will fix it later when finished with redux and sagas. I think I used **z-index** because the notes tree rendered with react-graph-vis will render on top of it (I think because it's rendered with canvas and after all react components are rendered).

Generally using **z-index** is not good, but in our case we just need a quick hack/fix.

You can use values like 2,3,4, etc., but it's hard to keep track of what is where. You can use numbers like 1000, but you quickly can run out of number.

Counting by hundreds seems like a reasonable choice. In our case we just want our button to be rendere above all other UI and markup.

File: src/components/App/AppComponent.js (hunk 6 above)
Comment (kvoncode) [reply src/components/App/AppComponent.js:24]: Yes. Totally agree. Will fix that

File: src/components/LoginPage/LoginPageComponent.js (hunk 2 above)
Comment (kvoncode) [reply src/components/LoginPage/LoginPageComponent.js:120]: **requestAuthorization** is dispatching action, so it's safe to assign it without checking gapi initialization.

File: src/components/LoginPage/LoginPageSagas.js (hunk 3 above)
Comment (kvoncode) [reply src/components/LoginPage/LoginPageSagas.js:37]: It's basically **console.log**

File: package.json [hunk 7]
```diff
@@ -4,7 +4,8 @@
   "description": "Note managment tool",
   "main": "server.js",
   "scripts": {
-    "start": "babel-node server/server.js",
+    "start": "webpack-dev-server --port 3000",
+    "start:old": "babel-node server/server.js",
```
Comment (DarthVanger) [inline package.json:8]: let's delete `start:old` and `server/server/js`? It looks like it's doing the same thing as the new `npm start`.

Comment (DarthVanger) [reply package.json:8]: And then we can delete `server/logger.js` as well

File: src/components/App/AppSagas.js [hunk 8]
```diff
@@ -11,11 +11,13 @@ import {
   CHANGE_PAGE_ACTION,
 } from 'components/App/AppActions';
 import { AUTH_SUCCESS_ACTION } from 'components/LoginPage/LoginPagesActions.js';
-import { PAGES_ENUM } from 'components/App/AppConstants';
+// import { PAGES_ENUM } from 'components/App/AppConstants';
```
Comment (DarthVanger) [inline src/components/App/AppSagas.js:14]: Do we still need `PAGES_ENUM` if we're using react router? Let's delete the enum?

File: src/components/App/AppSagas.js [hunk 9]
```diff
@@ -24,13 +26,14 @@ export function setPageAction(data) {
...
 
-  yield showSpinner('Loading Google Api');
+  // yield showSpinner('Loading Google Api');
   yield googleApiLoader.load();
   yield googleDriveApi.loadDriveApi();
   const rootNote = yield noteStorage.scanDrive();
   yield put(rootNoteFoundAction(rootNote));
-  yield setPageAction(PAGES_ENUM.NOTES);
-  yield hideSpinner();
+  // yield setPageAction(PAGES_ENUM.NOTES);
```
Comment (DarthVanger) [inline src/components/App/AppSagas.js:34]: Let's remove this commented code.

File: src/components/App/AppSagas.js [hunk 10]
```diff
@@ -24,13 +26,14 @@ export function setPageAction(data) {
...
+  yield put(push('/notes'));
+  // yield hideSpinner();
```
Comment (DarthVanger) [inline src/components/App/AppSagas.js:36]: Why removing spinner? If we don't need it, let's remove the code instead of commenting it.

File: src/components/App/AppSagas.js [hunk 11]
```diff
@@ -41,7 +44,8 @@ export function* appInit() {
     console.info('User is signed in');
     yield loadApp();
   } else {
-    yield takeEvery(AUTH_SUCCESS_ACTION, loadApp),
-      yield setPageAction(PAGES_ENUM.LOGIN);
+    yield takeEvery(AUTH_SUCCESS_ACTION, loadApp);
+    // yield put(push('/'));
```
Comment (DarthVanger) [inline src/components/App/AppSagas.js:48]: Why adding commented code?

File: src/components/App/AppSagas.js [hunk 12]
```diff
@@ -41,7 +44,8 @@ export function* appInit() {
...
+    // yield setPageAction(PAGES_ENUM.LOGIN);
```
Comment (DarthVanger) [inline src/components/App/AppSagas.js:49]: Let's remove this since we're using router now.

File: src/components/BottomBar/BottomBarComponent.js [hunk 13]
```diff
@@ -33,6 +37,51 @@ const StyledLabel = styled.div`
...
+
+      files.forEach(item => {
+        item.uploadFolderId = uploadFolderId;
+        item.abortController = new window.AbortController();
+      });
+
+      dispatch(UploadsActions.list.addedFiles(files));
+      fileInputRef.current.value = null;
+    }
+  }
```
Comment (DarthVanger) [inline src/components/BottomBar/BottomBarComponent.js:63]: What is this code for? Did you copy it from somewhere? How are uploads related to note details?

File: src/components/LoginPage/LoginPageComponent.js (hunk 2 above)
Comment (DarthVanger) [reply src/components/LoginPage/LoginPageComponent.js:120]: Saga listens to this action and calls `gapiAuthorize()` which calls `gapi.auth.authorize()`. So if `gapi` is not initialized, it will fail with `can not call .auth() of undefined`.
How does dispatching an action help us to make it safe?

File: src/components/LoginPage/LoginPageSagas.js (hunk 3 above)
Comment (DarthVanger) [reply src/components/LoginPage/LoginPageSagas.js:37]: It changes state
```
    case GOOGLE_API_INITIALIZED_ACTION:
      return Object.assign({}, state, { isGoogleApiInitialized: true });
```
so we can then know if `gapi` is already loaded or not. If it's not loaded we can't authorize the user: the user will click "login" and the app will break.

File: src/components/LogoutButton/LogoutButtonComponent.js [hunk 14]
```diff
@@ -1,12 +1,15 @@
 import React, { Component } from 'react';
-import { StyledLogoutButton } from 'components/LogoutButton/LogoutButtonStyles';
+// import { StyledLogoutButton } from 'components/LogoutButton/LogoutButtonStyles';
```
Comment (DarthVanger) [inline src/components/LogoutButton/LogoutButtonComponent.js:2]: I don't see anything useful inside `LogoutButtonStyles.js`. Let's delete it and let's delete this commented code.

File: src/components/NoteDetailsButton/NoteDetailsButtonComponent.js (hunk 5 above)
Comment (DarthVanger) [reply src/components/NoteDetailsButton/NoteDetailsButtonComponent.js:12]: aha I see, okay.

File: src/components/NoteDetailsPage/NoteDetailsPage.js [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+const BackButtonWrapper = styled.div`
+  color: red !important;
+  padding: 1rem;
+`;
+
+export const NoteDetailsPage = () => {
+  const dispatch = useDispatch();
+
+  const handleClick = () => {
+    dispatch({ type: CHANGE_PAGE_ACTION, data: PAGES_ENUM.NOTES });
```
Comment (DarthVanger) [inline src/components/NoteDetailsPage/NoteDetailsPage.js:24]: You have deleted code from `AppComponent.js` which used `state.page`, and now `react-router` is used. So why do we still need to dispatch `CHANGE_PAGE_ACTION`, which would update `state.page`, which is not used anywhere?

Comment (DarthVanger) [reply src/components/NoteDetailsPage/NoteDetailsPage.js:24]: Now we have two actions: `@@router/LOCATION_CHANGE` and `CHANGE_PAGE_ACTION`, so it's basically a logic duplication

image

Moreover, `CHANGE_PAGE_ACTION` is not changing `state` for this case, because the `NoteDetailsButton` never dispatches `CHANGE_PAGE_ACTION`. So when we go from `/notes` to `/note`, `state.page` stays as `/notes`, it never changes to `/note`. But then when we go back from `/note` to `/notes,` we try to update it to `/notes`... That's quite confusing.

If we want to replace our custom page change handling with `react-router`, let's delete everything related to `CHANGE_PAGE_ACTION`? What is the point of keeping both?

File: src/components/NotesMindMap/NotesMindMapComponent.js [hunk 16]
```diff
@@ -25,19 +25,49 @@ export class NotesMindMapComponent extends Component {
         keyboard: false,
       },
       edges: {
-        arrows: { to: true },
+        // arrows: { to: true },
```
Comment (DarthVanger) [inline src/components/NotesMindMap/NotesMindMapComponent.js:28]: Let's delete this commented code.

File: src/components/NotesMindMap/NotesMindMapComponent.js [hunk 17]
```diff
@@ -25,19 +25,49 @@ export class NotesMindMapComponent extends Component {
...
+        arrows: {
+          to: {
+            enabled: false,
+          },
+        },
+        color: '#F2F2F2',
+        // dashes: true,
```
Comment (DarthVanger) [inline src/components/NotesMindMap/NotesMindMapComponent.js:35]: Let's delete this commented code.

File: src/components/NotesMindMap/NotesMindMapComponent.js [hunk 18]
```diff
@@ -25,19 +25,49 @@ export class NotesMindMapComponent extends Component {
...
-          color: {
-            background: '#faa',
+          borderWidth: 2,
+          shape: 'box',
+          color: '#BB86FC',
+          margin: 10,
+          font: {
+            color: '#F2F2F2',
+            size: 15,
+            face: 'raleway',
```
Comment (DarthVanger) [inline src/components/NotesMindMap/NotesMindMapComponent.js:70]: We have a separate task to change the mind map rendering engine, and it's in progress: https://github.com/DarthVanger/NeuralNotes/issues/29

I like how the mind map looks with the new styles, but it's a waste of time... You're changing code which is going to be deleted anyway.

File: src/components/NotesPage/NotesPageComponent.js [hunk 19]
```diff
@@ -1,32 +1,24 @@
...
       <>
-        <TopBarComponent />
-        <OpenFileButtonComponent />
+        <TopBar />
         <NotesMindMapContainer />
-        <NotesContentEditorContainer />
-        {!isHelpViewed && <ControlsHelpComponent onClose={closeHelp} />}
-        <SearchPanelContainer />
-        <LogoutButtonContainer />
+        {/* {!isHelpViewed && <ControlsHelpComponent onClose={closeHelp} />} */}
```
Comment (DarthVanger) [inline src/components/NotesPage/NotesPageComponent.js:19]: Why did you remove help?

File: src/components/NotesPage/NotesPageComponent.js [hunk 20]
```diff
@@ -1,32 +1,24 @@
...
         <UploadStatusButton />
-        <UploadButton />
+        {/* <UploadButton /> */}
```
Comment (DarthVanger) [inline src/components/NotesPage/NotesPageComponent.js:21]: Why did you remove upload button?

File: src/components/TopBar/TopBar.js [hunk 21]
```diff
@@ -23,24 +25,17 @@ const useStyles = makeStyles(theme => ({
...
-    data,
-  };
-};
+const TopBarWrapper = styled.div`
+  display: flex;
+  height: 5rem;
+`;
 
-export function TopBarComponent() {
+export function TopBar() {
```
Comment (DarthVanger) [inline src/components/TopBar/TopBar.js:33]: Why renaming component name?
We could rename all components from `...Component` to `...`, if we agreed in that task were we discussed it.
But renaming only a few components, while leaving the other components with the old names is messy: you're breaking the convention, the code style. Now we're going to have `SearchPageComponent.js` but `TopBar.js`.

The [README](https://github.com/DarthVanger/NeuralNotes#structure) still says

>All files are using their type as filename postfix.

You're just making it more confusing for the other devs by renaming `TopBarComponent.js` to `TopBar.js`...

We should either change all the component names or change nothing... Otherwise it's a mess...

File: src/index.html [hunk 22]
```diff
@@ -1,19 +1,15 @@
...
       content="width=device-width, initial-scale=1, shrink-to-fit=no"
     />
 
     <link
-      href="https://fonts.googleapis.com/css?family=Raleway"
-      rel="stylesheet"
-    />
-    <link
-      href="https://fonts.googleapis.com/css?family=Roboto:300,400,700&display=swap"
+      href="https://fonts.googleapis.com/css?family=Raleway&family=ZCOOL+QingKe+HuangYou&&family=Roboto:300,400,700display=swap"
```
Comment (DarthVanger) [inline src/index.html:12]: Why do we need `family=ZCOOL+QingKe+HuangYou` ? @TomaTomoe are we using these fonts in our design?

The string `family=Roboto:300,400,700display=swap` is missing `&` between `700` and `display=swap`.

File: src/reducers.js [hunk 23]
```diff
@@ -7,11 +8,22 @@ import { searchPageReducer } from 'components/SearchPage/SearchPageReducer';
...
-  [uploadsReducer.KEY]: uploadsReducer,
-});
+// export const reducers = combineReducers({
+//   app: appReducer,
+//   notes: notesReducer,
+//   noteDetails: noteDetailsReducer,
+//   notesMindMap: notesMindMapReducer,
+//   searchPage: searchPageReducer,
+//   [uploadsReducer.KEY]: uploadsReducer,
+// });
```
Comment (DarthVanger) [inline src/reducers.js:18]: Please delete commented code.

File: src/sagas.js [hunk 24]
```diff
@@ -1,14 +1,18 @@
 import { all } from 'redux-saga/effects';
 import createSagaMiddleware from 'redux-saga';
-import { reducers } from 'reducers';
+import { createRootReducer } from 'reducers';
 import { applyMiddleware, createStore } from 'redux';
 import { composeWithDevTools } from 'redux-devtools-extension';
+import { createBrowserHistory } from 'history';
+// import { compose } from 'redux';
```
Comment (DarthVanger) [inline src/sagas.js:7]: Please delete commented code.

File: src/sagas.js [hunk 25]
```diff
@@ -1,14 +1,18 @@
...
+
+import { routerMiddleware } from 'connected-react-router';
 
 import { spinnerInit } from 'components/Spinner/SpinnerSagas';
 import { appInit } from 'components/App/AppSagas';
 import { loginInit } from 'components/LoginPage/LoginPageSagas';
 import { logoutButtonInit } from 'components/LogoutButton/LogoutButtonSagas';
-import { notesInit } from 'components/NotesPage/NotesPageSagas';
+// import { notesInit } from 'components/NotesPage/NotesPageSagas';
```
Comment (DarthVanger) [inline src/sagas.js:15]: Please delete commented code.

File: src/sagas.js [hunk 26]
```diff
@@ -33,10 +37,27 @@ const composeEnhancers = composeWithDevTools({
...
+
+  return store;
+};
+
+export const store = configureStore();
+
+// export const store = createStore(
+//   reducers,
+//   composeEnhancers(applyMiddleware(sagaMiddleware)),
+// );
```
Comment (DarthVanger) [inline src/sagas.js:60]: Please delete commendted code.

File: src/sagas.js [hunk 27]
```diff
@@ -48,7 +69,7 @@ export function* rootSaga() {
     spinnerInit(),
     loginInit(),
     logoutButtonInit(),
-    notesInit(),
+    // notesInit(),
```
Comment (DarthVanger) [inline src/sagas.js:72]: Why remove `notesInit()` ?

File: src/theme.js [hunk 28]
```diff
@@ -50,7 +50,7 @@ export const theme = createMuiTheme({
       fontStyle: 'normal',
       fontWeight: 500,
       fontSize: 14,
-      lineHeight: 16,
+      // lineHeight: 16, don't change line height as it's mess up markup
```
Comment (DarthVanger) [inline src/theme.js:53]: Which markup? What do you mean? @Batsvinova-Anastasiia , @TomaTomoe please review

File: webpack.config.js [hunk 29]
```diff
@@ -2,93 +2,99 @@ const path = require('path');
...
-const UglifyJsPlugin = require('uglifyjs-webpack-plugin')
+// const BundleAnalyzerPlugin = require('webpack-bundle-analyzer')
+//   .BundleAnalyzerPlugin;
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
 const CompressionPlugin = require('compression-webpack-plugin');
-const webpack = require('webpack')
+const webpack = require('webpack');
 
-const ASSET_PATH = process.env.ASSET_PATH || '/';
+// const ASSET_PATH = process.env.ASSET_PATH || '/';
```
Comment (DarthVanger) [inline webpack.config.js:11]: Why remove this?

Comment (DarthVanger) [reply webpack.config.js:11]: I think we never use public path different from `/`, so okay I guess we can remove this code...
Let's delete the commented code.

File: src/components/LoginPage/LoginPageComponent.js (hunk 2 above)
Comment (kvoncode) [reply src/components/LoginPage/LoginPageComponent.js:120]: If what you saying is true, the app would not work.
Screenshot 2020-10-25 07:32:01

**gapiAuthorize** are called independently of **isGoogleApiInitialized**

File: src/components/LoginPage/LoginPageSagas.js (hunk 3 above)
Comment (kvoncode) [reply src/components/LoginPage/LoginPageSagas.js:37]: We don't need it because auth code is run on `REQUEST_AUTHORIZATION_ACTION`, but `REQUEST_AUTHORIZATION_ACTION` only dispatched when `isGoogleApiInitialized: true`.

So basically we have a state field to change a click handler. Click handlers is safe to assign because they are not immediately executed.

File: package.json (hunk 7 above)
Comment (kvoncode) [reply package.json:8]: Yes. Will do that later when refactoring

File: src/components/App/AppSagas.js (hunk 8 above)
Comment (kvoncode) [reply src/components/App/AppSagas.js:14]: Ok. I leave the comments so it's immediately seen where code changes was made and when needed I can easily uncomment them to restore app working state, so please don't concentrate on deleting commented code, all comments will be deleted once all file/component are refactored. I regularly parsing through files so they will not be left undeleted

File: src/components/App/AppSagas.js (hunk 9 above)
Comment (kvoncode) [reply src/components/App/AppSagas.js:34]: Ok. I leave the comments so it's immediately seen where code changes was made and when needed I can easily uncomment them to restore app working state, so please don't concentrate on deleting commented code, all comments will be deleted once all file/component are refactored. I regularly parsing through files so they will not be left undeleted

File: src/components/App/AppSagas.js (hunk 10 above)
Comment (kvoncode) [reply src/components/App/AppSagas.js:36]: Ok. I leave the comments so it's immediately seen where code changes was made and when needed I can easily uncomment them to restore app working state, so please don't concentrate on deleting commented code, all comments will be deleted once all file/component are refactored. I regularly parsing through files so they will not be left undeleted

File: src/components/App/AppSagas.js (hunk 11 above)
Comment (kvoncode) [reply src/components/App/AppSagas.js:48]: So basically commented code replaces old code, but commenting it do not affect app. I found it interesting for figuring out why this happens later. Also if needed I can copy paste and then just change the route path.

File: src/components/App/AppSagas.js (hunk 12 above)
Comment (kvoncode) [reply src/components/App/AppSagas.js:49]: Please don't concentrate on refactoring and commented code. You can just checkout to merging branch and start app to see if everything working

File: src/components/BottomBar/BottomBarComponent.js (hunk 13 above)
Comment (kvoncode) [reply src/components/BottomBar/BottomBarComponent.js:63]: It's bottom bar functionality. It's not related to note page. 

It's good workflow to create branch and then regularly merge to master from one branch. So we if we have **note-page** branch:
- I add note page and then PR
- then you checkout to this branch and run app, check if it's working well
- then you review file changes and search for typos and mistakes
- if found comment on PR
- if not found merge and if some changes to be made related to this branch you comment continiously on merged branch 
- later I make changes and commit
- because it's same branch more merges can be made

Currently whats happening is that not any of my code goes to master, so the following problems occur

- rebase bugs. instead of rebase we could just merged and then fixed problem in later commits
- merge conflicts, everytime merge is happening conflicts are appearing, so it's good to merge often so the conflicts do not accumulate

Currently we can go forever with this branch. You will comment, I will make change and everytime master changes I will need resoleve more and more conflicts. So please merge and point out problems is new issues.

File: src/components/LogoutButton/LogoutButtonComponent.js (hunk 14 above)
Comment (kvoncode) [reply src/components/LogoutButton/LogoutButtonComponent.js:2]: Please leave that to refactoring branch

File: src/components/NoteDetailsPage/NoteDetailsPage.js (hunk 15 above)
Comment (kvoncode) [reply src/components/NoteDetailsPage/NoteDetailsPage.js:24]: Please leave that to refactoring branch

File: src/components/NotesMindMap/NotesMindMapComponent.js (hunk 16 above)
Comment (kvoncode) [reply src/components/NotesMindMap/NotesMindMapComponent.js:28]: Please leave that to refactoring branch

File: src/components/NotesMindMap/NotesMindMapComponent.js (hunk 17 above)
Comment (kvoncode) [reply src/components/NotesMindMap/NotesMindMapComponent.js:35]: Please leave that to refactoring branch

File: src/components/NotesMindMap/NotesMindMapComponent.js (hunk 18 above)
Comment (kvoncode) [reply src/components/NotesMindMap/NotesMindMapComponent.js:70]: I saw https://github.com/DarthVanger/NeuralNotes/issues/29 

I do not think it's wasted time because it just some styles, which do not take lot time. Also do not see reasons why `vis.js` can't be used

File: src/components/NotesPage/NotesPageComponent.js (hunk 19 above)
Comment (kvoncode) [reply src/components/NotesPage/NotesPageComponent.js:19]: There is no help in design, also If you need help is much easier to create new one that to make refactoring

File: src/components/NotesPage/NotesPageComponent.js (hunk 20 above)
Comment (kvoncode) [reply src/components/NotesPage/NotesPageComponent.js:21]: Upload button functionality are delegated to new design fab button. Please checkout to branch and run app to check application functionality

File: src/components/TopBar/TopBar.js (hunk 21 above)
Comment (kvoncode) [reply src/components/TopBar/TopBar.js:33]: I renaming components that I refactored so it's easier to distinguish them.

Please concentrate on making successfull app rather than fighting over convention or stylecode.

It's easier to read and write short names, and it's really remoralizing to write long names as <ComponentNameComponent>. I just try to do productive work, so you and your app are successful and I'm comfortable to do that work

File: src/reducers.js (hunk 23 above)
Comment (kvoncode) [reply src/reducers.js:18]: I need it untill redux, saga are completely refactored

File: src/sagas.js (hunk 27 above)
Comment (kvoncode) [reply src/sagas.js:72]:
```
export function* notesInit() {
  yield all([
    readHelpSeenState(),
    takeEvery(CHANGE_USER_HELP_SEEN_ACTION, saveHelpSeenState),
  ]);
}
```

It's better to use help on demand and just make intuitive UI

File: src/theme.js (hunk 28 above)
Comment (kvoncode) [reply src/theme.js:53]: Your master have visual bug with topbar. I fixed it. Just checkout to master before merge and run app

**Comment by kvoncode** [comment]
It's good workflow to create branch and then regularly merge to master from one branch. So we if we have note-page branch:

I add note page and then PR
then you checkout to this branch and run app, check if it's working well
then you review file changes and search for typos and mistakes
if found comment on PR
if not found merge and if some changes to be made related to this branch you comment continiously on merged branch
later I make changes and commit
because it's same branch more merges can be made
Currently whats happening is that not any of my code goes to master, so the following problems occur

rebase bugs. instead of rebase we could just merged and then fixed problem in later commits
merge conflicts, everytime merge is happening conflicts are appearing, so it's good to merge often so the conflicts do not accumulate
Currently we can go forever with this branch. You will comment, I will make change and everytime master changes I will need resoleve more and more conflicts. So please merge and point out problems is new issues.

File: src/components/LoginPage/LoginPageComponent.js (hunk 2 above)
Comment (kvoncode) [reply src/components/LoginPage/LoginPageComponent.js]: fixed by https://github.com/DarthVanger/NeuralNotes/pull/167/commits/2cf00e0e0593bd1f37ac43153229f3446b782603

File: src/components/LoginPage/LoginPageSagas.js (hunk 3 above)
Comment (kvoncode) [reply src/components/LoginPage/LoginPageSagas.js]: fixed by https://github.com/DarthVanger/NeuralNotes/pull/167/commits/2cf00e0e0593bd1f37ac43153229f3446b782603

File: src/index.html (hunk 22 above)
Comment (kvoncode) [reply src/index.html]: fixed by https://github.com/DarthVanger/NeuralNotes/pull/167/commits/af032cb32dd4e363a0163f3c851279751ea54f6b

File: src/theme.js (hunk 28 above)
Comment (kvoncode) [reply src/theme.js]: fixed by https://github.com/DarthVanger/NeuralNotes/pull/167/commits/cd171d7bfbb1e3684ed8c396b367f876df031843

File: src/components/NoteDetailsPage/NoteDetailsPage.js (hunk 15 above)
Comment (kvoncode) [reply src/components/NoteDetailsPage/NoteDetailsPage.js]: fixed by https://github.com/DarthVanger/NeuralNotes/pull/167/commits/26d2a17600286be2fa683cb0217232fe490d6fca

File: src/components/BottomBar/BottomBarComponent.js (hunk 13 above)
Comment (kvoncode) [reply src/components/BottomBar/BottomBarComponent.js:63]: My bad. Should have checked out to master and then made changes. When have merge conflicts please mention me I will resolve

File: src/sagas.js (hunk 27 above)
Comment (DarthVanger) [reply src/sagas.js:72]: How is `notesInit()` related to help?
